webpack/assets/javascripts/react_app/routes/ExternalRoutes.js

```diff
@@ -0,0 +1,25 @@
+import { Route } from 'react-router-dom';
+import React from 'react';
+import { addGlobalFill } from '../components/common/Fill/GlobalFill';
+import { renderRoute } from './RoutingService';
+
+/**
+ * Adds a plugin's routes into core
+ * @param  {String} id  plugin's id - can be its name
+ * @param  {Array}   routes an array that contains a plugin's routes
+ */
+export const registerRoutes = (id, routes) =>
```

Comment:
I find the name `ExternalRoutes.js` a bit confusing, would it make sense to change the terminology from `external routes` to `plugin routes`?

I think it will make the developer's life easier when they want to div deep into the code to understand how plugin routes work.